### PR TITLE
Update the CLP dependency to the lastest main branch commit

### PR DESCRIPTION
**Pull request by LinZhihao-723**
# Description
Update the CLP dependency to use the latest features implemented in the CLP core. This includes:
1. Use `BufferReader` to decode log events
2. Add protocol version validation when decoding the metadata to ensure the input ir stream is supported with the current library version.

# Validation performed
1. Passed all the unit tests.
2. Passed all the C++ linters.

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -1,5 +1,7 @@
+recursive-include src/clp/components/core/src *.h
 recursive-include src/clp/components/core/src *.hpp
 recursive-include src/clp/components/core/src *.tpp
+recursive-include src/clp/components/core/src *.inc
```
Comment (kirkrodrigues) [inline MANIFEST.in:3]: You no longer need the line above

Also, to make things consistent, do you want to do the following?
* Rename `src/clp_ffi_py/utils.tpp` to `.inc`
* Change the `clp_ffi_py *.tpp` rule below to `.inc`

File: src/clp_ffi_py/ir/native/decoding_methods.cpp [hunk 2]
```diff
@@ -181,6 +187,30 @@ auto decode_preamble(PyObject* Py_UNUSED(self), PyObject* py_decoder_buffer) ->
...
+                case ffi::ir_stream::IRProtocolErrorCode_Too_New:
+                    PyErr_Format(PyExc_RuntimeError, "Version too new: %s", version.c_str());
+                    break;
+                case ffi::ir_stream::IRProtocolErrorCode_Too_Old:
+                    PyErr_Format(PyExc_RuntimeError, "Version too old: %s", version.c_str());
+                    break;
+                default:
+                    PyErr_Format(
+                            PyExc_NotImplementedError,
+                            "Unrecognized return code %d with version value %s",
```
Comment (kirkrodrigues) [inline src/clp_ffi_py/ir/native/decoding_methods.cpp]:
```suggestion
                            "Unrecognized return code %d with version: %s",
```

File: src/clp_ffi_py/ir/native/decoding_methods.cpp [hunk 3]
```diff
@@ -181,6 +187,30 @@ auto decode_preamble(PyObject* Py_UNUSED(self), PyObject* py_decoder_buffer) ->
...
+                            error_code,
+                            version.c_str()
+                    );
```
Comment (kirkrodrigues) [inline src/clp_ffi_py/ir/native/decoding_methods.cpp:210]: Missing a `break`?

**Review by kirkrodrigues — CHANGES_REQUESTED** [review]

File: MANIFEST.in (hunk 1 above)
Comment (LinZhihao-723) [reply MANIFEST.in:3]: True, I've renamed .tpp to `.inc` inside `clp_ffi_py` to make them consistent.

**Review by kirkrodrigues — APPROVED** [review]

**Pull request merged**